mapie/classification.py

```diff
@@ -500,6 +523,138 @@ def _add_random_tie_breaking(
+        y_val_proba : ArrayLike
+            Probabilities of the validation set.
+
+        Returns
+        -------
+        ArrayLike
+            Probabilities with the right number of classes.
+        """
+        y_proba_tot = np.zeros(
+            shape=(len(y_proba), self.n_classes_)
+        )
+        y_index = np.tile(est_classes, (len(y_proba), 1))
+        np.put_along_axis(
+            y_proba_tot,
+            y_index,
+            y_proba,
+            axis=1
+        )
+        return y_proba_tot
+
+    def _predict_oof_model(
+        self,
+        estimator: ClassifierMixin,
+        X: ArrayLike,
+    ) -> ArrayLike:
+        """
+        Predict probabilities of a test set from a fitted estimator.
+
+        Parameters
+        ----------
+        estimator : ClassifierMixin
+            Fitted estimator.
+        X : ArrayLike
+            Test set.
+
+        Returns
+        -------
+        ArrayLike
+            Predicted probabilities.
+        """
+        y_pred_proba = estimator.predict_proba(X)
+        # we enforce y_pred_proba to contain all labels included y
+        if len(estimator.classes_) != self.n_classes_:
+            y_pred_proba = self._fix_number_of_classes(
+                estimator.classes_,
+                y_pred_proba
+            )
+        return y_pred_proba
+
+    def _fit_and_predict_oof_model(
+        self,
+        estimator: ClassifierMixin,
+        X: ArrayLike,
+        y: ArrayLike,
+        train_index: ArrayLike,
+        val_index: ArrayLike,
+        k: int,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> Tuple[ClassifierMixin, ArrayLike, ArrayLike, ArrayLike]:
+        """
+        Fit a single out-of-fold model on a given training set and
+        perform predictions on a test set.
+
+        Parameters
+        ----------
+        estimator : ClassifierMixin
+            Estimator to train.
+
+        X : ArrayLike of shape (n_samples, n_features)
+            Input data.
+
+        y : ArrayLike of shape (n_samples,)
+            Input labels.
+
+        train_index : np.ndarray of shape (n_samples_train)
+            Training data indices.
+
+        val_index : np.ndarray of shape (n_samples_val)
+            Validation data indices.
+
+        k : int
+            Split identification number.
+
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights. If None, then samples are equally weighted.
+            By default None.
+
+        Returns
+        -------
+        Tuple[ClassifierMixin, ArrayLike, ArrayLike, ArrayLike]
+
+        - [0]: Fitted estimator
+        - [1]: Estimator predictions on the validation fold,
+          of shape (n_samples_val,)
+        - [2]: Identification number of the validation fold,
+          of shape (n_samples_val,)
+        - [3]: Validation data indices,
+          of shape (n_samples_val,).
+
+        """
+        X_train, y_train, X_val, y_val = (
```

Comment:
I think you can proceed without `y_val`

Reply:
`y_val` is used for creating `val_id` at the end of the function.